### ISSUE TITLE
Simplify dependencies

### ISSUE THREAD

**Issue by kadykov** [MLResearchAtOSRAM/tmm_fast#18]
Since the `tmm-fast` is published on [pipy.org](https://pypi.org/project/tmm-fast/) I think it is better to keep dependencies as minimal as possible.

For instance, it would be better to remove `dask` from dependencies, as it is not directly used anywhere in the project.

The main functionality in [vectorized_tmm_dispersive_multistack.py](https://github.com/MLResearchAtOSRAM/tmm_fast/blob/main/tmm_fast/vectorized_tmm_dispersive_multistack.py) and [tmm_fast_torch.py](https://github.com/MLResearchAtOSRAM/tmm_fast/blob/main/tmm_fast/tmm_fast_torch.py) does not require `gymnasium`, `matplotlib` and `seaborn`, therefore we can declare them as optional ones.

Probably, it could be even better to split the TMM and AI functionality into two separate projects...

**Comment by Nerrror** [comment]
You're right, it might be a good idea to split the TMM and AI functionality and add a second project which then builds on TMM for the RL environment.
This could be implemented in the next main version or so. I'm working on an extension for incoherent layers right now so this issue might take a while to implement
